src/main/java/net/coderbot/patchwork/manifest/forge/AccessTransformerList.java

```diff
@@ -0,0 +1,59 @@
+package net.coderbot.patchwork.manifest.forge;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.objectweb.asm.commons.Remapper;
+
+public class AccessTransformerList {
+
+	private List<AccessTransformerEntry> entries;
+	public AccessTransformerList(List<AccessTransformerEntry> entries) {
+		this.entries = entries;
+	}
+
+	public List<AccessTransformerEntry> getEntries() {
+		return entries;
+	}
+
+	// TODO fails mysteriously when unable to remap
+	public static AccessTransformerList parse(Path accessTransformer) throws Exception {
+		List<String> lines = Files.exists(accessTransformer) ?
+									 Files.readAllLines(accessTransformer) :
+									 new ArrayList<>();
+		Map<String, String> ats = new HashMap<>(); // map of AT classes and fields/methods
+		// ATs are formatted like this:
+		// public-f com/example/Foo bar # mars
+		// We can make everything public and definalized and let the mod just live in its own world,
+		// so we only use the 2nd and 3rd words.
+		for(String line : lines) {
+			// Put everything into the map. Changes package "." to folder "/".
+			// regex.
+			String[] split = line.replaceAll("\\.", "/").split(" ");
+			// If the line is greater than three words AND none of those words contains a comment
+			// symbol
+			if(!(split.length < 3 || (split[0] + split[1] + split[2]).contains("#"))) {
+				ats.put(split[1], split[2]);
+			}
+		}
+		// for every AT the mod uses
+		List<AccessTransformerEntry> entries = new ArrayList<>();
```

Comment:
You put stuff into a map only to put it back into a list, this could be 
```
		List<AccessTransformerEntry> entries = new ArrayList<>();
		for (String line : lines) {
			String[] split = line.replaceAll("\\.", "/").split(" ");
			if (!(split.length < 3 || (split[0] + split[1] + split[2]).contains("#"))) {
				entries.add(new AccessTransformerEntry(split[1], split[2]));
			}
		}
```
And skip the map ordeal.